compiler/tests-compiler/space_optimization.ml

```diff
@@ -0,0 +1,130 @@
+      [%expect {|
+        e1 || e2;
+        //end|}])
+
+let%expect_test "one branch if to && expression statement" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "if (e1) e2;" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        e1 && e2;
+        //end|}])
+
+let%expect_test "expression statement; var declaration" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "e1; var x = e2;" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        var x = (e1, e2);
+        //end|}])
+
+let%expect_test "expression statement; return statement" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "e1; return e2;" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        return e1, e2;
+        //end|}])
+
+let%expect_test "expression statement; if statement" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "e1; if (e2) 0; else 1;" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        e1, e2 ? 0 : 1;
+        //end|}])
+
+let%expect_test "expression statement; if statement without else" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "e1; if (e2) 0;" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        e1, e2 && 0;
+        //end|}])
+
+let%expect_test "if statement without else; return statement" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "if (e1) e2; return e3" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        return e1 && e2, e3;
+        //end|}])
+
+let%expect_test "expression statement; expression statement" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "e1; e2;" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        e1, e2;
+        //end|}])
+
+let%expect_test "\"use strict\"; expression statement" =
+  with_temp_dir ~f:(fun () ->
+      let js_prog = "\"use strict\"; e2;" in
+      let js_file =
+        js_prog |> Filetype.js_text_of_string |> Filetype.write_js ~name:"test.js"
+      in
+      print_endline (Util.optimize_space js_file);
+      [%expect {|
+        "use strict"; e2;
+        //end|}])
```

Comment:
Maybe add a check that `if (e1) e2; if (e3) e4 else e5; return e6` becomes
`return e1 && e2, e3 ? e4 : e5, e6`
and similarly for `var` instead of return?
That is, checking that more than one statements can be pushed into the one following.

Reply:
[398a1b9](https://github.com/ocsigen/js_of_ocaml/pull/1467/commits/398a1b9fbc16ac66b00a8833f6e40a082f089640)